### DISCUSSION TITLE
Cache fallback for Trivy in client/server mode with a (failing) Redis cache?

### DISCUSSION THREAD

**Discussion by elchenberg** [aquasecurity/trivy#3309] [Q&A] [answered]
I run Trivy in client/server mode with a Redis cache. When a Trivy client scans a container image while the cache backend is unavailable the scan fails.

```sh
trivy --version
Version: 0.35.0

# Trivy server (for demonstration purposes the Redis cache backend does not exist)
trivy server --cache-backend redis://localhost:6379 --cache-ttl 1h --listen localhost:4954

# Trivy client
trivy image --server http://localhost:4954 busybox:latest
```

The error message looks like this:

```
FATAL	image scan error: scan error: scan failed: failed analysis: analyze error: failed to store layer: sha256:c9fba744b45ded00d622ce131a1bd5ca3a71ad7dea44ccf9ffbe3f6d8d6122a3 in cache: unable to store blob information in Redis cache (sha256:c9fba744b45ded00d622ce131a1bd5ca3a71ad7dea44ccf9ffbe3f6d8d6122a3): dial tcp 127.0.0.1:6379: connect: connection refused
```

Is it possible to make the error non-fatal? Maybe Trivy could use the filesystem cache as a fallback while Redis is unavailable? Is there some configuration that I am missing or would this be a new feature?

**Comment by knqyf263** [comment, marked as answer]
Trivy fails on the Redis connection refusing and doesn't have an option to fall back into other cache options.

**Reply by elchenberg** [reply]
Thank you for the prompt reply!

In my case I want Redis cache failures it to be non-fatal. I will have a look. Maybe I can come up with a solution.
